prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -517,14 +518,26 @@ class ActionHistory(
         filename: String?,
         params: String,
         result: String,
-        itemCount: Int
+        header: WorkflowEngine.Header
     ) {
         if (isReportAlreadyTracked(sentReportId)) {
             error(
                 "Bug:  attempt to track history of a report ($sentReportId) " +
                     "we've already associated with this action"
             )
         }
+
+        if (header.content == null) {
+            error("Bug: attempt to track sent report with no contents")
+        }
+
+        val blobInfo = BlobAccess.uploadBody(
```

Comment:
This is the forward looking change so that we can later remove the code that handles the ReportFile from the send step not having a `bodyurl`